libs/npm/core/src/hooks/api/interfaces/IUserProductModel.ts

```diff
@@ -0,0 +1,10 @@
+import { IUserModel } from '.';
+
+export interface IUserProductModel extends IUserModel {
+  /** Whether the user is subscribed to this product. */
+  isSubscribed: boolean;
+  /** Whether the user has requested to be subscribed. */
+  requestedIsSubscribedStatus?: boolean; // TODO: I find this status confusing
```

Comment:
naming things is hard - I'm open to suggestions
- `requestedIsSubscribedStatus` : Does the user want to subscribe or unsubscribe - Scott apparently wants to know either way.  This value in combination with the `isSubscribed` can be used so that we can infer "They are subscribed but they want to unsubscribe"
- `subscriptionChangeActioned` : Has an admin actioned this request.  Used to show/infer the status of the subscription on both the subscriber and admin page